File: app/src/main/java/me/imzack/app/ender/injector/component/PlanSearchComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.PlanSearchPresenterModule;
import me.imzack.app.ender.injector.scope.ActivityScope;
import me.imzack.app.ender.view.activity.PlanSearchActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = PlanSearchPresenterModule.class)
public interface PlanSearchComponent {

    void inject(PlanSearchActivity planSearchActivity);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/PlanSearchComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.PlanSearchPresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.PlanSearchActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(PlanSearchPresenterModule::class))
interface PlanSearchComponent {

    fun inject(planSearchActivity: PlanSearchActivity)
}
